packages/core-data/src/hooks/use-entity-record.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * WordPress dependencies
+ */
+import { useQuerySelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { store as coreStore } from '../';
+import { Status } from './constants';
+
+interface EntityRecordResolution< RecordType > {
+	/** The requested entity record */
+	record: RecordType | null;
+
+	/**
+	 * Is the record still being resolved?
+	 *
+	 * Note that isResolving and hasResolved may both be true when the entity record is being re-requested.
+	 */
+	isResolving: boolean;
+
+	/**
+	 * Is the record resolved by now?
+	 *
+	 * Note that isResolving and hasResolved may both be true when the entity record is being re-requested.
+	 */
+	hasResolved: boolean;
+
+	/** Resolution status */
+	status: Status;
+}
+
+/**
+ * Resolves the specified entity record.
+ *
+ * @param {string} kind     Kind of the requested entity.
```

Comment:
I believe we have a lint rule somewhere that prefers typescript's type syntax over JSDoc syntax, so this should be an error. Not sure why it isn't working in this file 🤔 . I think it makes sense so that we don't have to declare the same types twice.

Reply:
I just adjusted it